src/libraries/Microsoft.Extensions.Hosting.Abstractions/ref/Microsoft.Extensions.Hosting.Abstractions.cs

```diff
@@ -38,10 +38,10 @@ public static partial class Environments
     }
     public partial class HostBuilderContext
     {
-        public HostBuilderContext(System.Collections.Generic.IDictionary<object, object> properties) { }
-        public Microsoft.Extensions.Configuration.IConfiguration Configuration { get { throw null; } set { } }
-        public Microsoft.Extensions.Hosting.IHostEnvironment HostingEnvironment { get { throw null; } set { } }
-        public System.Collections.Generic.IDictionary<object, object> Properties { get { throw null; } }
+        public HostBuilderContext(System.Collections.Generic.IDictionary<object, object?> properties) { }
+        public Microsoft.Extensions.Configuration.IConfiguration? Configuration { get { throw null; } set { } }
+        public Microsoft.Extensions.Hosting.IHostEnvironment? HostingEnvironment { get { throw null; } set { } }
+        public System.Collections.Generic.IDictionary<object, object?> Properties { get { throw null; } }
```

Comment:
My gut tells me that this should be `IDictionary<object, object> Properties`. Looking at how ASP.NET uses it, they don't expect `null` to be put into this dictionary. I guess if you really wanted `null`, you should just remove the key from the dictionary, or don't put it in there in the first place.

So I'd say let's go with not making the values nullable here.